ls1intum/Artemis
# `Tutorial groups`: Fix an issue with the configuration

### Checklist
#### General


#### Server

### Motivation and Context

In the client code certain entities are sent to the server with an indirect connection to the tutorial group configuration. The problem was that the setters for the tutorial group configuration had a restriction that required dates to be in the 'uuuu-MM-dd' format. The Jackson library was used for deserialization of these entities, but it was failing because the indirectly connected configuration was sending dates in the full ISO string format, which is the default for the dayjs library. This mismatch between the expected date format and the actual date format was causing Jackson to fail the entire deserialization process. 

To fix this bug, the code was updated to move the date format checks out of the setters, allowing dates to be sent in the full ISO string format. This allowed Jackson to successfully deserialize the entities, and the problem was resolved.

### Steps for Testing
- One way to replicate the bug that was found by @ge65cer :
- Create a course and configure tutorial groups (create one or two) to make sure that the configuration exists in the database
- Reload the whole page to clear the client cache
- Now try to create a programming exercise. This should work without any problems now.

#### Exam Mode Testing
- Same as above but with an exam programming exercises

### Review Progress

#### Code Review
#### Manual Tests
#### Exam Mode Test

### Test Coverage
Added tests that show the error is fixed now.
